File: src/main/scala/mffs/security/module/ItemModuleAntiHostile.java
package mffs.security.module;

import resonant.api.mffs.security.IInterdictionMatrix;
import mffs.ModularForceFieldSystem;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.entity.INpc;
import net.minecraft.entity.monster.IMob;

public class ItemModuleAntiHostile extends ItemModuleInterdictionMatrix
{
	public ItemModuleAntiHostile(int i)
	{
		super(i, "moduleAntiHostile");
	}

	@Override
	public boolean onDefend(IInterdictionMatrix interdictionMatrix, EntityLivingBase entityLiving)
	{
		if (entityLiving instanceof IMob && !(entityLiving instanceof INpc))
		{
			entityLiving.attackEntityFrom(ModularForceFieldSystem.damagefieldShock, 20);
		}

		return false;
	}
}

File: src/main/scala/mffs/security/module/ItemModuleAntiHostile.scala
package mffs.security.module

import java.util

import mffs.api.machine.Projector
import nova.core.entity.component.Damageable
import nova.core.util.transform.vector.Vector3i

class ItemModuleAntiHostile extends ItemModuleDefense
{

  override def onCreateField(projector: Projector, field: util.Set[Vector3i]): Boolean = {
    val entities = getEntitiesInField(projector)

    //Check entity IDs.
    entities.view
        .filter(entity => entity.isInstanceOf[Damageable] /* && entity.isInstanceOf[IMob] && !entity.isInstanceOf[INpc]*/)
        .map(_.asInstanceOf[Damageable])
        .foreach(_.damage(20))

    return false
  }

    override def getID: String = "moduleAntiHostile"
}